source/material_model/simple_nonlinear.cc

```diff
@@ -0,0 +1,435 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/material_model/simple_nonlinear.h>
+#include <aspect/utilities.h>
+#include <aspect/newton.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace
+  {
+    std::vector<double>
+    get_vector_double (const std::string &parameter, const unsigned int n_fields, ParameterHandler &prm)
+    {
+      std::vector<double> parameter_list;
+      parameter_list = Utilities::string_to_double(Utilities::split_string_list(prm.get (parameter)));
+      if (parameter_list.size() == 1)
+        parameter_list.resize(n_fields, parameter_list[0]);
+
+      AssertThrow(parameter_list.size() == n_fields,
+                  ExcMessage("Length of " + parameter + " list (size "+ std::to_string(parameter_list.size()) +") must be either one,"
+                             " or n_compositional_fields+1 (= " + std::to_string(n_fields) + ")."));
+
+      return parameter_list;
+    }
+  }
+
+  namespace MaterialModel
+  {
+    template <int dim>
+    std::vector<double>
+    SimpleNonlinear<dim>::
+    compute_volume_fractions( const std::vector<double> &compositional_fields) const
+    {
+      /*std::cout << "vfFlag 1 " << std::endl;
+         std::vector<double> volume_fractions(compositional_fields.size());
+         std::cout << "vfFlag 2 " << std::endl;
```

Comment:
if this debug output is still helpful, it would be worthwhile writing it in the style of properly indented code. But I suspect that you can get rid of it.